charts/argo-cd/templates/redis/secret.yaml

```diff
@@ -0,0 +1,60 @@
+# lookup existing secret
+{{- $secretName :=  include "argo-cd.redis.fullname" . -}}
+{{- $secretObj := (lookup "v1" "Secret" .Release.Namespace $secretName) | default dict }}
```

Comment:
Using lookup inside the Argo ecosystem is IMHO a bad idea. Argo CD uses helm only as a template engine (argocd only executes `helm template ...`).
And `helm template` does not support looking up values for security reasons.

Reply:
Hi @mkilchhofer, Thanks for bringing up this topic! If we're talking about ArgoCD in general, I fully agree. However, in this particular case the Helm Chart is meant for bootstrapping ArgoCD. Basically, this can be achieved in two ways:
1. Install directly via Helm -> Lookup works
2. ArgoCD manages ArgoCD -> `ignoreDifferences` for the password fields needs to be specified

I would really love to see a more secure default installation of ArgoCD. Especially, because not all clusters support `NetworkPolicies` out of the box (e.g. Flannel doesn't). 

I suggest to add an example for `ignoreDifferences` and some explanation to the README to address the questions arising by using lookup. WDY?

Reply:
I mean chaning the default installation without being in-line with with upstrean kustomize manifests is not what we want.
Upstream is our reference for the default installation as mentioned in the charts README.

I appreciate a more secure default installation but it has to be promoted first as a PR in the upstream repo (https://github.com/argoproj/argo-cd).

Reply:
According to the Chart's README:
> Hence, it makes sense to try to achieve a similar output of rendered .yaml resources when calling helm template using the default settings in values.yaml

According to this, the default should be `redis.auth.enabled=false` (as it is in the current implementation). Additionally, a user can easily enable a more secure installation by setting `redis.auth.enabled` to `true`. 
Sounds reasonable to me.